messages.go

```diff
@@ -139,21 +145,34 @@ type TreeNodeInfo struct {
 type OverlayMsg struct {
 	TreeNodeInfo *TreeNodeInfo
 
-	RequestTree   *RequestTree
-	TreeMarshal   *TreeMarshal
+	// Deprecated: roster is not sent/requested anymore, only the tree
 	RequestRoster *RequestRoster
-	Roster        *Roster
+	// Deprecated: roster is not sent/requested anymore, only the tree
+	Roster *Roster
+
+	RequestTree  *RequestTree
+	ResponseTree *ResponseTree
+	// Deprecated: use ResponseTree to send the tree and the roster
+	TreeMarshal *TreeMarshal
+}
+
+// RequestRoster is used to ask the parent for a given Roster
+type RequestRoster struct {
+	RosterID RosterID
 }
 
 // RequestTree is used to ask the parent for a given Tree
 type RequestTree struct {
 	// The treeID of the tree we want
 	TreeID TreeID
+	// Version of the request tree
+	Version uint32
```

Comment:
Should we make a list of all these ugly backward compatibility code somewhere so that they can be removed in v4?

Reply:
I actually put the `Deprecated: ` thing 1.) to explain and 2.) make an easy searchable tag to remove. Now it might be interesting to have a list of what else can be removed if we remove that maybe.